langserver/langserver_test.go

```diff
@@ -1216,16 +1219,16 @@ func lspTests(t testing.TB, ctx context.Context, fs *AtomicFS, c *jsonrpc2.Conn,
 		wantGodefHover = cases.wantHover
 	}
 
-	if len(wantGodefDefinition) > 0 || (len(wantGodefHover) > 0 && fs != nil) || len(cases.wantCompletion) > 0 {
-		UseBinaryPkgCache = true
+	if len(wantGodefDefinition) > 0 || (len(wantGodefHover) > 0 && h != nil) || len(cases.wantCompletion) > 0 {
```

Comment:
shouldn't you be checking that `h.FS != nil`

Reply:
Prior to this PR, it seems `fs` being `nil` was a signal to not perform the tests in the `if` block.  See [integration_test.go](https://github.com/pbitty/go-langserver/blob/add-config-struct/langserver/integration_test.go#L76-L137).

In this case, we're passing the `LangHandler` instead of `FS`, but approach remains the same.  The way I see it, I sorta hacked around the issue and it is now a little bit harder to read.

I'm happy to submit another PR to refactor these specific tests.  Maybe we can break out this block into its own func with a clear name, and only call it where it makes sense?

Reply:
I'm happy to accept any PR you send which cleans stuff up :) Checking `h != nil` LGTM